checks/system/integration/utils.py

```diff
@@ -0,0 +1,213 @@
+        elif '*' not in include:
+            if self.CLASS not in include:
+                return
+
+        #
+        # If where is unspecificed we can run with any feature.
+        # Be forgiving if the user forgets a leading +.
+        #
+
+        if not where:
+            where = '*'
+        elif where[0] not in ['-', '+']:
+            where = f'+{where}'
+
+        #
+        # Get out properties ready.
+        #
+
+        name                = f'Check_{Check.check_id:04}{lcl_CHECK_CLASS}'
+        valid_systems       = where.split()
+        valid_prog_environs = ['builtin']
+        time_limit          = '2m'
+
+        if self.DEBUG:
+            exp_str, nexp_str = '', ''
+            if expected is not None:
+                exp_str  = f"expected:{expected}"
+            if not_expected is not None:
+                nexp_str = f"not_expected:{not_expected}"
+
+            print(f"{debuginfo()} {name:25} {cmd:80} -> {exp_str} {nexp_str}" +
+                  (f"  on  {valid_systems} with {valid_prog_environs}"))
+            return
+
+        #
+        # Wait until here to do any imports so that if we are running in debug
+        # mode we don't depend on the ReFrame framework being present.
+        #
+
+        import reframe as rfm
+        import reframe.utility.sanity as sn
+        import reframe.core.builtins as builtins
+
+        from reframe.core.meta import make_test
+
+        def validate(test):
+            """Callback to check that we got the output we expected."""
+
+            a, b = True, True
+
+            expected, where = test.expected
+            if expected is not None:
+                where = eval(f'test.{test.expected[1]}')
+                a = sn.assert_found(
+                    expected,
+                    where,
+                    msg=f"Expected '{expected}' running" +
+                        f"'{test.cmd} {test.caller}'")
+
+            not_expected, where = test.not_expected
+            if not_expected is not None:
+                where = eval(f'test.{test.not_expected[1]}')
+                b = sn.assert_not_found(
+                    not_expected,
+                    where,
+                    msg=f"Did not expect '{not_expected}'" +
+                        f"running '{test.cmd} {test.caller}'")
+            return (a and b)
+
+        def set_command_options(test):
+            """Set up the options we need to run"""
+
+            test.executable = test.cmd
+
+            if isinstance(test.expected, list):
+                test.expected = test.expected
+            else:
+                test.expected = [test.expected, 'stdout']
+
+            if isinstance(test.not_expected, list):
+                test.not_expected = test.not_expected
+            else:
+                test.not_expected = [test.not_expected, 'stdout']
+
+            test.skip_if(
+                test.expected[1] not in ['stdout', 'stderr'],
+                msg=f'Location for expected' +
```

Comment:
```suggestion
                msg=f'Location for expected ' +
```
Tiny nitpick, leave an empty space before `'`, so that the last word is not joined with the first word of the next line. Same for the other `msg` in the file.

Reply:
Good catch. I split all the strings because the lines were more than 80 characters and didn't notice this.